spack.yaml

```diff
@@ -5,26 +5,35 @@
 spack:
   # add package specs to the `specs` list
   specs:
-    - access-om2@git.2024.03.0
+    - access-om2@git.2024.06.0
   packages:
     cice5:
-      require: '@git.2023.10.19'
+      require:
+        - '@git.2023.10.19'
     mom5:
-      require: '@git.2023.11.09'
+      require:
+        - '@git.2023.11.09'
     libaccessom2:
-      require: '@git.2023.10.26'
+      require:
+        - '@git.2023.10.26'
     oasis3-mct:
-      require: '@git.2023.11.09'
+      require:
+        - '@git.2023.11.09'
     netcdf-c:
-      require: '@4.7.4'
+      require:
+        - '@4.9.2'
     netcdf-fortran:
-      require: '@4.5.2'
+      require:
+        - '@4.6.1'
     parallelio:
-      require: '@2.5.2'
+      require:
+        - '@2.6.2'
     openmpi:
-      require: '@4.0.2'
+      require:
+        - '@4.1.5'
     all:
-      compiler: [intel@19.0.5.281]
+      compiler: [intel@2021.10.0]
```

Comment:
Use `intel-oneapi-compilers-classic@2021.6.0` for the Issue that was opened to update to the latest dependencies/compilers. For CI (not CD), openmpi needs to be compiled and will fail with `intel-oneapi-compilers-classic@2021.10.0`.

Reply:
Does the version formulation `intel-oneapi-compilers-classic@2021.6.0` work for `gadi` (CD) ad GitHub runners (CI)?

Reply:
That's the compiler package name and version. The compiler name and version are `intel@2021.6.0`

Reply:
For CI, this spec works: `openmpi@4.1.5 cppflags="-diag-disable=10441" %intel@2021.10.0 target=x86_64`

Reply:
I'm not too sure. That might be a @harshula question...

Reply:
Just so I understand - this comment is about the CI compiler versioning, not anything to do with the current formulation of the spack.yaml?

Reply:
This spec builds openmpi with `intel@2021.10.0`: `openmpi@4.1.5 cppflags="-diag-disable=10441" %intel@2021.10.0 target=x86_64`

Reply:
We want to keep the CI and CD compilers in-sync. Historically we didn't because we only had 2021.x available for CI and ACCESS-OM2 had been verified with 2019.x.

Reply:
So what you're saying is downgrade the compiler to `2021.6.0` for CD because `2021.10.0` doesn't build on CI?

Reply:
Two options to choose from:
* Use `2021.6.0` for both CI/CD
* Use `2021.10.0` and build `openmpi` with the following spec for CI: `openmpi@4.1.5 cppflags="-diag-disable=10441" %intel@2021.10.0 target=x86_64`

Reply:
I think option 2 would be better. I don't think what we can do on Gadi for deployment should be gated by what we can do in the image. 
I'm also considering reworking `build-ci` when the majority of the `*-configs` infrastructure is matured, so it can easily accept those kind of flags as input. That repo was one of the first things I'd worked on and I think it needs a proper freshen up in order for it to be useful to people in the dev team, rather than bolting on features.

Reply:
OK, let's go with option 2. Moving forward, we need to use the same compiler version for CI and CD. Can you please test what happens to CD when `cppflags="-diag-disable=10441"` is added to the `spack.yaml`? You may have to review the schema and investigate the best way to incorporate it. Alternatively, I can try to incorporate it into the `openmpi` SPD of our Spack fork.

Reply:
We can definitely incorporate that into the spack.yaml. Will do so in a sec and let you know.

Reply:
@harshula it has been deployed successfully as `pr60-12`

Reply:
On Gadi, it looks like the system `openmpi` was used, as we want. I'm compiling the model from this `spack.yaml` in a dev container to double check that the `cppflags` are applied.

Reply:
`spack-build-env.txt` contained:
```
export SPACK_CPPFLAGS=-diag-disable=10441;
```